server/meshmodel/helper.go

```diff
@@ -0,0 +1,137 @@
+package meshmodel
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+
+	"github.com/layer5io/meshery/server/helpers/utils"
+	"github.com/layer5io/meshery/server/models"
+	"github.com/layer5io/meshkit/models/meshmodel"
+	"github.com/layer5io/meshkit/models/meshmodel/core/v1alpha1"
+	"github.com/pkg/errors"
+)
+
+const ArtifactHubComponentsHandler = "kubernetes" //The components generated in output directory will be handled by kubernetes
+
+type ComponentHelper struct {
+	handlerConfig    *models.HandlerConfig
+	regManager       *meshmodel.RegistryManager
+	componentChan    chan v1alpha1.ComponentDefinition
+	relationshipChan chan v1alpha1.RelationshipDefinition
+	doneSignal       chan bool
+	errorChan        chan error
+}
+
+func NewComponentHelper(hc *models.HandlerConfig, rm *meshmodel.RegistryManager) *ComponentHelper {
+	return &ComponentHelper{
+		handlerConfig:    hc,
+		regManager:       rm,
+		componentChan:    make(chan v1alpha1.ComponentDefinition, 1),
+		relationshipChan: make(chan v1alpha1.RelationshipDefinition, 1),
+		doneSignal:       make(chan bool),
+		errorChan:        make(chan error),
+	}
+}
+
+// seed the local meshmodel components
+func (ch ComponentHelper) SeedComponents() {
+	// Watch channels and register components and relationships with the registry manager
+	go ch.watchComponents()
+
+	// Read component and relationship definitions from files and send them to respective channels
+	ch.generateComponents("/components")
+	ch.generateRelationships("/relationships")
+	ch.doneSignal <- true
+}
+
+func (ch ComponentHelper) generateComponents(pathToComponents string) {
```

Comment:
Please make these methods for Pointer to ComponentHelper. You are lucky that all variables are effectively pointers so a shallow copy didn't hurt. But please make it a pointer to the methods, especially when you are communicating values in the same instance of the struct.